central/networkgraph/aggregator/aggregator_impl.go

```diff
@@ -316,3 +327,14 @@ func normalizeDupNameExtSrcs(entity *storage.NetworkEntityInfo) {
 		},
 	}
 }
+
+// Return NetworkEntityInfo_INTERNET iif:
+// - entity is a 'discovered' external entity
+// - displaying of 'discovered' entities in the network-graph is disabled
+// Otherwise, return entity.
+func anonymizeIfNoExternalIPs(entity *storage.NetworkEntityInfo, externalIPs bool) *storage.NetworkEntityInfo {
+	if networkgraph.IsExternalDiscovered(entity) && !externalIPs {
+		return networkgraph.InternetEntity().ToProto()
+	}
+	return entity
+}
```

Comment:
If the other comment is applied, then we can simplify the behavior of this one a bit
```suggestion
func anonymizeIfNoExternalIPs(entity *storage.NetworkEntityInfo) *storage.NetworkEntityInfo {
	if networkgraph.IsExternalDiscovered(entity) {
		return networkgraph.InternetEntity().ToProto()
	}
	return entity
}
```

that would make it much easier to test.

Reply:
You may want to also rename this function, as the if-part would be moved outside the function body

Reply:
Indeed! I renamed to reflect that this affects only discovered external entities.